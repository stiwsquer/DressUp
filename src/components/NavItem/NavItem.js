import React, { useContext } from "react";
import { BrowserRouter as NavLink, Link } from "react-router-dom";
export default function NavItem({
  linkTo,
  iconClass,
  text,
  onClickFunction,
  reference,
}) {
  return (
    // <li onClick={onClickFunction} ref={reference}>
    <Link to={linkTo}>
      <i className={iconClass}></i>
      {iconClass !== null && " "}
      {text}
    </Link>
    // </li>
  );
}
