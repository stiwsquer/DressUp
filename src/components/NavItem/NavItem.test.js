import NavItem from "./NavItem";

it("renders NavItem component without crashing", () => {
  shallow(<NavItem linkTo="#" />);
});

it("renders one Link component", () => {
  const wrapper = shallow(<NavItem linkTo="#" />);
  expect(wrapper.find("Link")).toHaveLength(1);
});

it("has proper className", () => {
  const wrapper = shallow(<NavItem linkTo="#" iconClass="fa-bars" />);
  expect(wrapper.find("i").first().hasClass("fa-bars")).toBe(true);
  expect(wrapper).toMatchSnapshot();
  wrapper.setProps({ iconClass: "fa-times" });
  expect(wrapper.find("i").first().hasClass("fa-times"));
  expect(wrapper).toMatchSnapshot();
});

it("renders proper text", () => {
  const wrapper = shallow(<NavItem linkTo="#" text="proper text" />);
  expect(wrapper.find("Link").first().text()).toEqual("proper text");
  expect(wrapper).toMatchSnapshot();
  wrapper.setProps({ text: "text" });
  expect(wrapper.find("Link").first().text()).toEqual("text");
  expect(wrapper).toMatchSnapshot();
});
