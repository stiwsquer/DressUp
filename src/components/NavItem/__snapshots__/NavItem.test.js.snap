// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has proper className 1`] = `
<Link
  to="#"
>
  <i
    className="fa-bars"
  />
   
</Link>
`;

exports[`has proper className 2`] = `
<Link
  to="#"
>
  <i
    className="fa-times"
  />
   
</Link>
`;

exports[`renders proper text 1`] = `
<Link
  to="#"
>
  <i />
  proper text
</Link>
`;

exports[`renders proper text 2`] = `
<Link
  to="#"
>
  <i />
  text
</Link>
`;
